include/boost/gil/extension/numeric/convolve.hpp

```diff
@@ -169,6 +169,16 @@ void convolve_cols(const SrcView& src, const Kernel& ker, const DstView& dst,
     convolve_rows<PixelAccum>(transposed_view(src),ker,transposed_view(dst),option);
 }
 
+/// \ingroup ImageAlgorithms
+///convolve a 1D variable-size kernel along both rows and columns of an image
+template <typename PixelAccum,typename SrcView,typename Kernel,typename DstView>
+BOOST_FORCEINLINE
+void convolve(const SrcView& src, const Kernel& ker, const DstView& dst,
+                   convolve_boundary_option option=convolve_option_extend_zero) {
```

Comment:
Nitpicking:

```
) {
```
is preferred to be
```
)
{
```

Such big indentations are not necessary:

```
                   convolve_...
```

Finally, I'd call for new lines to follow the [consistent const](http://slashslash.info/petition/) (aka [East const](https://mariusbancila.ro/blog/2018/11/23/join-the-east-const-revolution/)) :-)